src/datepicker/datepicker.component.ts

```diff
@@ -307,7 +307,7 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 	 * Handles opening the calendar "properly" when the calendar icon is clicked.
 	 */
 	openCalendar(datepickerInput: DatePickerInput) {
-		datepickerInput.input.nativeElement.click();
+		this.flatpickrInstance.open();
```

Comment:
@Donisius I believe you changed this last ... do you remeber why we switched to `datepickerInput.input.nativeElement.click` from `this.flatpickrInstance.open`?

Reply:
The months do not get updated properly when the calendar is opened through `this.flatpickrInstance.open` when the calendar is in range mode but it needs to be opened through `this.flatpickrInstance.open` when the datepicker is in single mode otherwise it will not open when the calendar icon is clicked. @yinglejia Sorry, it was perfect the way it was originally.

Reply:
OK, I'll revert the last commit then.